test/index.ts

```diff
@@ -541,71 +541,94 @@ function testDeepNonNullable() {
 
 function testDeepRequire() {
   type cases = [
-    Assert<IsExact<DeepRequired<number | null | undefined>, number>>,
-    Assert<IsExact<DeepRequired<string | null | undefined>, string>>,
-    Assert<IsExact<DeepRequired<boolean | null | undefined>, boolean>>,
-    Assert<IsExact<DeepRequired<bigint | null | undefined>, bigint>>,
-    Assert<IsExact<DeepRequired<symbol | null | undefined>, symbol>>,
-    Assert<IsExact<DeepRequired<undefined | null>, never>>,
-    Assert<IsExact<DeepRequired<Function | null | undefined>, Function>>,
-    Assert<IsExact<DeepRequired<Date | null | undefined>, Date>>,
-    Assert<IsExact<DeepRequired<Error | null | undefined>, Error>>,
-    Assert<IsExact<DeepRequired<RegExp | null | undefined>, RegExp>>,
-    Assert<IsExact<DeepRequired<Map<string | null | undefined, boolean | null | undefined>>, Map<string, boolean>>>,
+    Assert<IsExact<DeepRequired<number | null | undefined>, number | null | undefined>>,
```

Comment:
The tests are broken from the first commit

You can see that all of the test cases are not valid now – https://github.com/krzkaczor/ts-essentials/runs/3415539355

It will be fixed with the next commit